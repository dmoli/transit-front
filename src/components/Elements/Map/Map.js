/* global GOOGLE_MAPS_ID */
import { compose, withProps, withState, withHandlers } from 'recompose';
import React, { Component } from 'react';
import {
  withScriptjs,
  withGoogleMap,
  GoogleMap,
  Polyline,
} from 'react-google-maps';

// Google Maps v3 API Options Reference:
// https://developers.google.com/maps/documentation/javascript/3.exp/reference
const defaultOptions = {
  mapTypeControl: false,
  zoomControl: true,
  streetViewControl: true,
  fullscreenControl: false,
  scrollwheel: false,
};

const MapWithControlledZoom = compose(
  withProps({
    googleMapURL: `https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=${GOOGLE_MAPS_ID}`,
    loadingElement: <div style={{ height: '100%' }} />,
    containerElement: <div style={{ height: '100%' }} />,
    mapElement: <div style={{ height: '100%' }} />,
  }),
  withState('zoom', 'onZoomChange', 15),
  withHandlers(() => {
    const refs = {
      map: undefined,
    };

    return {
      onMapMounted: () => (ref) => {
        refs.map = ref;
      },
      onZoomChanged: ({ onZoomChange }) => () => {
        onZoomChange(refs.map.getZoom());
      },
    };
  }),
  withScriptjs,
  withGoogleMap,
)(props =>
  <GoogleMap
    center={props.center}
    zoom={props.zoomCustom}
    ref={props.onMapMounted}
    onZoomChanged={props.onZoomChanged}
    defaultOptions={defaultOptions}
  >
    <Polyline
      path={props.markers}
      options={{
        geodesic: true,
        strokeColor: '#e9004b',
        strokeOpacity: 1.0,
        strokeWeight: 2,
      }}
    ></Polyline>
  </GoogleMap>,
);

/**
 * Clase contenedora de Google Maps
 */
class Map extends Component {
  render() {
    return (
      <MapWithControlledZoom
        {...this.props}
      />
    );
  }
}

export default Map;
