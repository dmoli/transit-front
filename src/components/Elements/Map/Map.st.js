/* eslint import/no-extraneous-dependencies: [0] */
import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';

import Map from './Map';

const center = { lat: -33.4314474, lng: -70.6093325 };
const markers = [
  {
    lat: -33.4314474,
    lng: -70.6093325,
    show: true,
  },
  {
    lat: -33.4533624,
    lng: -70.714299,
    show: true,
  },
];

storiesOf('Elements/Map', module)
  .add('Macardores', () => (
    <Container>
      <Map
        center={center}
        markers={markers}
        zoomCustom={17}
      />
    </Container>
  ));

const Container = styled.div`
  width: 50%;
  height: 400px;
`;
