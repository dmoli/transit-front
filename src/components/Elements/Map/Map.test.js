/* eslint-env jest */
import React from 'react';
import { mount } from 'enzyme';

import Map from './Map';

const center = { lat: -33.4314474, lng: -70.6093325 };
const markers = [
  {
    lat: -33.4314474,
    lng: -70.6093325,
    show: true,
  },
  {
    lat: -33.4533624,
    lng: -70.714299,
    show: true,
  },
];
describe('<Map />', () => {
  it('should rendered correctly', () => {
    const wrapper = mount(
      <Map
        center={center}
        markers={markers}
        zoomCustom={17}
      />,
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
