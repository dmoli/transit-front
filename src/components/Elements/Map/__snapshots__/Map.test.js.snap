// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> should rendered correctly 1`] = `"<div style=\\"height: 100%;\\"></div>"`;
